js/settingsHelpers.js

```diff
@@ -964,3 +964,46 @@ export const unlinkFirebaseAuthenticationTrigger = async (authToUnlink) =>  {
       throw error;
   }
 }
+
+export const formerNameOptions = [
+  { value: "", text: "-- Select --" , i18n: 'form.formerNameCategoryDefaultOption'},
+  { value: "first", text: "First" , i18n: 'form.formerNameCategoryOption.first'},
+  { value: "middle", text: "Middle" ,i18n: 'form.formerNameCategoryOption.middle'},
+  { value: "last", text: "Last" ,i18n: 'form.formerNameCategoryOption.last'},
+];
+export const numberOfDefaultFormerNames = 2
+export const getFormerNameData = () => {
+    const formerNameItems = document.getElementsByClassName("former-name-item");
+    const fNameArray = [];
+    const mNameArray = [];
+    const lNameArray = [];
+    Array.from(formerNameItems).forEach((_, index) => {
+        const inputElement = document.getElementById(
+            `former-name-value-${index + 1}`
+        );
+        const selectElement = document.getElementById(
+            `former-name-category-${index + 1}`
+        );
+        if (inputElement.value) {
+            switch (selectElement.value) {
+                case "first":
+                    fNameArray.push(inputElement.value);
+                    break;
+                case "middle":
+                    mNameArray.push(inputElement.value);
+                    break;
+                case "last":
+                    lNameArray.push(inputElement.value);
+                    break;
+            }
+        }
+    });
+    return fNameArray.length || mNameArray.length || lNameArray.length
+        ? {
+              [cId.userProfileUpdateTimestamp]: new Date().toISOString(),
+              [cId.fName]: fNameArray.length ? fNameArray : undefined,
+              [cId.mName]: mNameArray.length ? mNameArray : undefined,
+              [cId.lName]: lNameArray.length ? lNameArray : undefined,
+          }
+        : undefined;
+};
```

Comment:
I really think we need to match these data structures for long-term compatibility.

Here's a screenshot of the data strucure this generated (from Firestore dev):

<img width="516" alt="Screenshot 2024-11-20 at 4 10 54 PM" src="https://github.com/user-attachments/assets/f8391a34-7947-4d26-9653-68683e509351">

---
Here's an example of the existing data structure for the user profile history spec:
<img width="580" alt="Screenshot 2024-11-20 at 4 12 18 PM" src="https://github.com/user-attachments/assets/a7a08d5a-3542-413d-8234-8a86ceb5a32a">

Reply:
Here is the result after using the code change.
<img width="364" alt="image" src="https://github.com/user-attachments/assets/1cb700cf-8653-4d79-9305-7782c16682ab">

Reply:
I see mixed data issues in the `[Array]` parts. Those are strings in the existing UP history implementation.

We are starting to use that UP history for some things like NORC Birthday cards, and we need to make sure the data is in a predictable format.

There will be more entries in the array, but I think those should be expanded out so each entry for 231676651, 399159511, and 996038075 is a separate object in the 569151507 UP History array.

Each UP history object needs:
•update string (first/middle/last name).
•update timestamp 371303487.
•profile change requested by 611005658 (this will be the participant's email address for this case).

Reply:
Just updated, please check it again.